src/v1/lib/json.ts

```diff
@@ -347,9 +348,11 @@ export function JSONEnum<K, T extends JSONObject>(enumDef: K): (target: T, prope
 export function JSONParsable(value: TypeStruct = {}) {
   return function def<T extends { new (...args: any[]): JSONObject }>(constructor: T) {
     return class StaticThis extends constructor implements JSONObject {
+      // @ts-ignore
       ___jsontypes___: TypeStruct = this.___jsontypes___
-        ? Object.assign(Object.assign({ ...this.___jsontypes___ }, this.___jsonprops___), value)
-        : Object.assign({ ...(this.___jsonprops___ || {}) }, value);
+        // @ts-ignore
+        ? ({...this.___jsontypes___, ...this.___jsonprops___, ...value})
+        : ({...(this.___jsonprops___ || {}), ...value});
```

Comment:
@rockon999 do you have context why this can pass TS a while ago? Now it's showing `used before initialized` error